### PR TITLE
Use `OS.kernel_name`

### PR DESCRIPTION
**Pull request by carlocab**
Needs Homebrew/brew#12014 to land in a release tag.

### CODE REVIEW

File: Formula/hasura-cli.rb [hunk 1]
```diff
@@ -35,12 +35,8 @@ def install
     end
 
     cd "cli" do
-      arch = Hardware::CPU.arm? ? "arm64" : "amd64"
-      os = if OS.mac?
-        "darwin"
-      else
-        "linux"
-      end
+      arch = Hardware::CPU.intel? ? "amd64" : Hardare::CPU.arch.to_s
```
Comment (carlocab) [inline Formula/hasura-cli.rb:38]:
```suggestion
      arch = Hardware::CPU.intel? ? "amd64" : Hardware::CPU.arch.to_s
```
Oops

File: Formula/logstash.rb [hunk 2]
```diff
@@ -47,13 +47,8 @@ def install
 
     # Delete Windows and other Arch/OS files
     rm Dir["bin/*.bat"]
-    os = if OS.mac?
-      "Darwin"
-    else
-      "x86_64-Linux"
```
Comment (dawidd6) [inline Formula/logstash.rb:53]: This architecture prefix is not included in `OS.kernel_name` output I think?

File: Formula/lsof.rb [hunk 3]
```diff
@@ -29,7 +29,7 @@ def install
 
       "darwin"
     else
-      "linux"
```
Comment (dawidd6) [inline Formula/lsof.rb:32]: You can pull the above logic into separate block for macOS.

**Comment by carlocab** [comment]
I'll just drop `openshift-cli` from here and re-run this as syntax-only so we can merge the reset ASAP.

**Pull request merged**